lib/posix-process/tests/test_clone.c

```diff
@@ -0,0 +1,140 @@
+	 *                     ^ 
+	 *                     | STACK 
+	 *                     v 
+	 * stack --------------------------------------------------------------- 
+	 *                         ^           ^ 
+	 *                         |           | TLS SPACE 
+	 *                         |           v 
+	 * ukplat_tlsp_get(),tcb ->|-TLS AREA------------------------------------
+	 *                         |           ^
+	 *                         |           | LIBC TCB (e.g pthread structure)
+	 *                         v           v
+	 *----------------------------------------------------------------------
+	 *
+	 *
+	 * Layout of the mapping aarch64
+	 * (there is also some padding but we omit it):
+	 * map ----------------------------------------------------------------
+	 *                    ^ 
+	 *                    |           STACK 
+	 *                    v 
+	 * stack --------------------------------------------------------------
+	 *                  tcb -> ^           ^ 
+	 *                         |           | LIBC tcb (e.g pthread structure) 
+	 *                         |           v 
+	 *      ukplat_tlsp_get()->|-TLS AREA-----------------------------------
+	 *                         |           ^
+	 *                         |           | TLS SPACE
+	 *                         v           v
+	 *--------------------------------------------------------------------
+	 */
+
+	size_t stack_size = 8 * 4096; // not using the default size
+	size_t tls_size = ukarch_tls_area_size();
+	__u8 *map = mmap(NULL, stack_size + tls_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+	__clone_child_tls_area = (unsigned long) map + stack_size;
+	void *tcb = __uk_copy_tls((void*) __clone_child_tls_area); // function defined in __uk_init_tls.c glue src file (musl)
+	void *stack = map +  stack_size;
+	int ctid;
+
+	main_thread_tid = syscall(SYS_gettid);
+
+	/* these flags are used by pthread_create() when a new thread is createad */
+	unsigned flags = CLONE_VM | CLONE_FS | CLONE_FILES | CLONE_SIGHAND
+                | CLONE_THREAD | CLONE_SYSVSEM | CLONE_SETTLS
+                | CLONE_PARENT_SETTID | CLONE_CHILD_CLEARTID | CLONE_DETACHED;
+
+	__clone_child_tls_ptr = (unsigned long) ukarch_tls_tlsp((void *) __clone_child_tls_area);
+	/* Use the clone wrapper from musl.
+	 * NB: this will call SYS_exit after clone_thread_fun() returns
+	 */
+	__clone(clone_thread_fun, stack, flags, tcb, &ctid, __clone_child_tls_ptr, &ctid);
+	/* clone will force the execution of child after it creates it,
+	 * but just to be sure the child executes and finishes,
+	 * we place some sleeps here.
+	 */
+	unsigned int no_of_tries = 5;
+	while (no_of_tries--) {
+		if (!__check_if_child_exited(__clone_child_uk_thread_ptr))
+			sleep(1);
+		else
+			break;
+	}
+
+	if (!__check_if_child_exited(__clone_child_uk_thread_ptr)) {
+		PRINT_BAD("VERY BAD the child didn't exit, let's commit suicide");
```

Comment:
Please take this out as well, because is no longer defined and the compilation will fail.